From 8125d84d97ed71ec86fb122591ec7f436af9ba80 Mon Sep 17 00:00:00 2001
From: Spike Liao <spike.liao@amlogic.com>
Date: Tue, 10 Feb 2015 19:03:06 +0800
Subject: [PATCH 5419/5965] PD#103277: h264: fix frame rate check bug

if sync_outside is set,always trust IDRFrame pts and duration

Change-Id: Ib7d708f32d73705bcb14cf9217d3db816d30f536
---
 drivers/amlogic/amports/vh264.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/amlogic/amports/vh264.c b/drivers/amlogic/amports/vh264.c
index fe48194e135f..4651203ede80 100755
--- a/drivers/amlogic/amports/vh264.c
+++ b/drivers/amlogic/amports/vh264.c
@@ -1266,7 +1266,7 @@ static void vh264_isr(void)
 
             last_pts = pts;
 
-            if (fixed_frame_rate_flag && (fixed_frame_rate_check_count <= FIX_FRAME_RATE_CHECK_IDRFRAME_NUM)) {
+            if (fixed_frame_rate_flag && (fixed_frame_rate_check_count <= FIX_FRAME_RATE_CHECK_IDRFRAME_NUM) && (sync_outside == 0)) {
                 pts = pts_lookup_save;
             }
 
-- 
2.19.0

